benchmarks/newton_solver_benchmark_set/spiegelman_et_al_2016/drucker_prager_compositions.cc

```diff
@@ -0,0 +1,711 @@
+          prm.declare_entry ("Maximum viscosity", "1e28", Patterns::List(Patterns::Double(0)),
+                             "Upper cutoff for effective viscosity. Units: $Pa s$");
+          prm.declare_entry ("Effective viscosity coefficient", "1.0", Patterns::List(Patterns::Double(0)),
+                             "Scaling coefficient for effective viscosity.");
+          prm.declare_entry ("Reference viscosity", "1e22", Patterns::List(Patterns::Double(0)),
+                             "Reference viscosity for nondimensionalization. Units $Pa s$");
+          prm.declare_entry ("Reference compressibility", "4e-12", Patterns::Double (0),
+                             "The value of the reference compressibility. Units: $1/Pa$.");
+
+          // averaging parameters
+          prm.declare_entry ("Viscosity averaging p", "-1",
+                             Patterns::Double(),
+                             "This is the p value in the generalized weighed average eqation: "
+                             " mean = \\frac{1}{k}(\\sum_{i=1}^k \\big(c_i \\eta_{\\text{eff}_i}^p)\\big)^{\\frac{1}{p}}. "
+                             " Units: $Pa s$");
+
+          // finite difference versus analytical
+          prm.declare_entry ("Use analytical derivative", "false",
+                             Patterns::Bool(),
+                             "A bool indicating wether to use finite differences to compute the derivative or to use "
+                             "the analytical derivative.");
+
+          prm.declare_entry ("Use deviator of strain-rate", "true",
+                             Patterns::Bool(),
+                             "Todo.");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+
+
+    template <int dim>
+    void
+    DruckerPragerCompositions<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      // can't use this->n_compositional_fields(), because some
+      // tests never initiate the simulator, but uses the material
+      // model directly.
+      prm.enter_subsection ("Compositional fields");
+      {
+        n_fields = prm.get_integer ("Number of fields")+1;
+        //AssertThrow(n_fields > 0, ExcMessage("This material model needs at least one compositional field."))
+        reference_T_list = get_vector_double("List of reftemps",n_fields,prm);//
+        ref_visc_list = get_vector_double ("List of initial viscosities",n_fields,prm);//
+        thermal_diffusivity = get_vector_double("List of conductivities",n_fields,prm);
+        heat_capacity = get_vector_double("List of capacities",n_fields,prm);
+
+        // ---- Compositional parameters
+        densities = get_vector_double("List of refdens",n_fields,prm);
+        thermal_expansivities = get_vector_double("Thermal expansivities",n_fields,prm);
+
+        // Rheological parameters
+        cohesion = get_vector_double("List of cohesions",n_fields,prm);
```

Comment:
```suggestion
        cohesion = get_vector_double("Cohesions",n_fields,prm);
```

Reply:
maybe change `cohesion` to `cohesions` to make it clear it is a vector of values rather than a single value?